nixos/modules/services/networking/pangolin.nix

```
@@ -0,0 +1,567 @@
+      http_entrypoint = "web";
+      https_entrypoint = "websecure";
+    };
+    gerbil = {
+      start_port = 51820;
+      base_endpoint = cfg.dashboardDomain;
+      block_size = 24;
+      site_block_size = 30;
+      subnet_group = "100.89.128.1/24";
+      use_subdomain = true;
+    };
+    rate_limits.global = {
+      window_minutes = 1;
+      max_requests = 500;
+    };
+    users.server_admin = {
+      email = cfg.letsEncryptEmail;
+      password = "Password123!"; # this is a dummy password and is overwritten by the environment variable USERS_SERVERADMIN_PASSWORD declared in pangolinEnvironmentFile
+    };
+    flags = {
+      require_email_verification = false;
+      disable_signup_without_invite = true;
+      disable_user_create_org = true;
+      allow_raw_resources = true;
+      allow_base_domain_resources = true;
+    };
+  };
+in
+{
+  options = {
+    services.pangolin = {
+      enable = lib.mkEnableOption "Pangolin";
+      package = lib.mkPackageOption pkgs "pangolin" { };
+
+      settings = lib.mkOption {
+        type = with lib.types; attrsOf str;
+        default = { };
+        description = ''
+          Additional attributes to be merged with the configuration options and written to Pangolin's config.yml file.
+        '';
+        example = {
+          app = {
+            save_logs = true;
+          };
+          domains.domain1 = {
+            prefer_wildcard_cert = true;
+          };
+        };
+      };
+      openFirewall = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = ''
+          Whether to open ports 80, 443, and 51820 in the firewall for the Pangolin service(s).
+        '';
+      };
+
+      baseDomain = lib.mkOption {
+        type = lib.types.str;
+        default = "example.com";
+        description = ''
+          Your base fully qualified domain name (without any subdomains).
+        '';
+        example = "example.com";
+      };
+      dashboardDomain = lib.mkOption {
+        type = lib.types.str;
+        default = "pangolin.${cfg.baseDomain}";
+        defaultText = "pangolin.\${config.services.pangolin.baseDomain}";
+        description = ''
+          The domain where the application will be hosted. This is used for many things, including generating links. You can run Pangolin on a subdomain or root domain. Do not prefix with http or https.
+        '';
+        example = "pangolin.example.com";
+      };
+
+      letsEncryptEmail = lib.mkOption {
+        type = with lib.types; nullOr str;
+        default = config.security.acme.defaults.email;
+        defaultText = lib.literalExpression "config.security.acme.defaults.email";
+        description = ''
+          An email address for SSL certificate registration with Lets Encrypt. This should be an email you have access to.
+        '';
+      };
+      # provide path to file to keep secrets out of the nix store
+      pangolinEnvironmentFile = lib.mkOption {
```

Why prepending `pangolin` ?

```suggestion
      environmentFile = lib.mkOption {
```

---

gerbil has its own environment option, and want to keep the two distinct at first glance

---

I wanted to recommend moving gerbil-related options into `services.gerbil` or something, which is enabled by `services.pangolin`.
But I felt it was nitpicking at this point, so I left that for a later refactor.

---

Will change to gerbilEnvironmentFile, since it looks like traefik is gonna need one as well. Will also change this in Newt

---

Going back to this issue, please put the gerbil options under `service.gerbil`. Don't prefix options with the service name.